### DISCUSSION TITLE
Capturing localhost traffic

### DISCUSSION THREAD

**Discussion by Eonasdan** [justcoding121/Titanium-Web-Proxy#886] [Q&A] [answered]
I've used the Basic, WPF and a new console app following the readme. I don't see any traffic from localhost. I'm developing a web app and I'd like to be able to capture and log api calls from my angular -> api.

**Comment by justcoding121** [comment, marked as answer]
Localhost traffic is tricky ? most browsers bypass the system proxy for `localhost` / `127.0.0.1` on purpose.

Workarounds:
1. Hit your app via the machine name / LAN IP instead of `localhost` (e.g. `http://192.168.x.x:4200`).
2. Chrome: launch with `--proxy-server=127.0.0.1:8000` and use a hostname that isn't in the bypass list. You can also try `http://local.mydomain` mapped in hosts.
3. Firefox: set `network.proxy.allow_hijacking_localhost` to `true` in about:config.
4. Don't rely on `SetAsSystemProxy` alone for loopback ? configure the client explicitly.

Also make sure Angular/`HttpClient` isn't talking straight to the API without going through the browser proxy settings.

Closing ? reopen if a specific browser still skips the proxy after the above.
